connectors/sources/onedrive.py

```diff
@@ -276,10 +275,7 @@ async def get(self, url):
             else:
                 raise
 
-    async def paginated_api_call(self, url_name, **url_kwargs):
-        url = parse.urljoin(BASE_URL, ENDPOINTS[url_name].format(**url_kwargs))
-        url = f"{url}?$top={FETCH_SIZE}"
-
+    async def paginated_api_call(self, url):
```

Comment:
Just one nit, I do think the method paginated_api_call should allow callers to specify the `fetch_size`.

```suggestion
    async def paginated_api_call(self, url, params=None, fetch_size=FETCH_SIZE):
        if params is None:
            params = {}
        params["$top"] = fetch_size
        params = "&".join(f"{key}={val}" for key, val in params.items())
        url = url + "?" + params # should apply url escape here?
```

Reply:
You're right. Updated with commit [f8cce96](https://github.com/elastic/connectors-python/pull/1619/commits/f8cce9667efb0fc7a29eb7ae454f3189c857dc0b)